oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/AuthorizablePropertiesImpl.java

```diff
@@ -234,9 +236,34 @@ private PropertyState getAuthorizableProperty(@NotNull Tree authorizableTree, @N
         }
         ReadOnlyNodeTypeManager nodeTypeManager = authorizable.getUserManager().getNodeTypeManager();
         PropertyDefinition def = nodeTypeManager.getDefinition(parent, property, true);
-        if (def.isProtected() || (authorizablePath.equals(parent.getPath())
-                && !def.getDeclaringNodeType().isNodeType(UserConstants.NT_REP_AUTHORIZABLE))) {
+        if (def.isProtected()) {
+            // exclude all protected properties
             return null;
+        } else if (authorizablePath.equals(parent.getPath())) {
+            // non-protected properties on the root must be defined by the expected
+            //  primary type or one of the configured mixin types
+            Boolean allowed = null;
```

Comment:
i would change that to 
`boolean allowed = false;`
and then set it to `allowed = true` as i don't see any benefit of a 'null' value.
what was the reason for this?

Reply:
Sure, I'll make that cleanup change as the current impl only needs true or false.  This code was leftover from a failed experiment where a third state would have been relevant.